doc/cloud-init.md

```diff
@@ -1,12 +1,19 @@
-# Custom network configuration with cloud-init
+# cloud-init
```

Comment:
I think the updated doc should at least:
 - Link to https://cloudinit.readthedocs.io/en/latest/topics/examples.html
 - Mention the relationship between vendor and user data and suggest vendor data be used in profiles
 - Probably mention `cloud-init status`?

Reply:
I updated the doc but didn't mention `cloud-init status`. Why should it be included? It only mentions whether it was successful, and the data source.